src/org/jitsi/service/neomedia/MediaStream.java

```diff
@@ -585,4 +585,43 @@ public void setRTCPTerminationStrategy(
      *
      */
     boolean isStartOfFrame(byte[] buf, int off, int len);
+
+    /**
+     * Utility method that determines whether or not a packet is an end of
+     * frame.
+     *
+     * @param buf the buffer that holds the RTP payload.
+     * @param off the offset in the buff where the RTP payload is found.
+     * @param len then length of the RTP payload in the buffer.
+     *
+     * @return true if the packet is the end of a frame, false otherwise.
+     *
+     * FIXME(gp) conceptually this belongs to the {@link VideoMediaStream}, but
+     * I don't want to be obliged to cast to use this method.
+     *
+     */
+    public boolean isEndOfFrame(byte[] buf, int off, int len);
+
+    /**
+     * Utility method that determines the temporal layer index (TID) of an RTP
+     * packet.
+     *
+     * @param buf the buffer that holds the RTP payload.
+     * @param off the offset in the buff where the RTP payload is found.
+     * @param len then length of the RTP payload in the buffer.
+     *
+     * @return the TID of the packet, -1 otherwise.
+     *
+     * FIXME(gp) conceptually this belongs to the {@link VideoMediaStream}, but
+     * I don't want to be obliged to cast to use this method.
+     */
+    public int getTemporalID(byte[] buf, int off, int len);
+
+    /**
+     * Gets the {@link MediaStreamTrackReceiver} of this {@link MediaStream}.
+     *
+     * @return the {@link MediaStreamTrackReceiver} of this {@link MediaStream},
+     * or null.
+     */
+    public MediaStreamTrackReceiver getMediaStreamTrackReceiver();
```

Comment:
I am bothered by the additions of all of these to the public and generic MediaStream interface. I have the following suggestion:
1. remove isEndOfFrame and (for the time being) use RawPacket.isPacketMarked() directly
2. Remove isStartOfFrame from MediaStream. Leave it only in MediaStreamImpl. Make MediaStreamTrackReceive aware of MediaStreamImpl, and have it return a MediaStreamImpl from getSteam(). 
3. For getTemporalId -- the same as above.
4. Similar for getMediaStreamTrackReceiver -- make the two classes which use it (both under impl.neomedia) aware that they have a MediaStreamImpl, and remove the method from the interface.

This adds a dependency on MediaStreamImpl to a couple of classes in impl.neomedia, but I think this is better than cluttering the public MediaStream interface. Of course, I'm not aware of what future changes are planned, so I might be missing usages.